ospackage/recipes/steps.go

```diff
@@ -84,39 +64,56 @@ func StepFileExec(step *osconfigpb.SoftwareRecipe_Step_FileExec, artifacts map[s
 		var ok bool
 		path, ok = artifacts[v.ArtifactId]
 		if !ok {
-			return nil, fmt.Errorf("%q not found in artifact map", v.ArtifactId)
+			return fmt.Errorf("%q not found in artifact map", v.ArtifactId)
 		}
 	default:
-		return nil, fmt.Errorf("can't determine location type")
+		return fmt.Errorf("can't determine location type")
 	}
 
-	res := []string{path}
-	res = append(res, step.FileExec.Args...)
-	return res, nil
+	return executeCommand(path, filepath.Join(runDir, "stepName"), runEnvs, step.FileExec.Args...)
 }
 
 // StepScriptRun builds the command for a ScriptRun step
-func StepScriptRun(step *osconfigpb.SoftwareRecipe_Step_ScriptRun, artifacts map[string]string) ([]string, error) {
+func StepScriptRun(step *osconfigpb.SoftwareRecipe_Step_ScriptRun, artifacts map[string]string, runEnvs []string, runDir string) error {
 	// TODO: should be putting this in stepN_type/ dir, but that needs me to know the dir way in advance..
 	// actually this is an artifact. we should have made them referenced as artifacts.. we'll have to repeat file creation logic here.
 	f, err := os.Create("/tmp/scriptrun")
 	if err != nil {
-		return nil, err
+		return err
 	}
 	defer f.Close()
 
 	f.WriteString(step.ScriptRun.Script)
 	f.Sync()
 	if err := os.Chmod("/tmp/scriptrun", 0755); err != nil {
-		return nil, err
+		return err
 	}
 
-	res := []string{"/bin/sh", "-c"}
-	//if step.ScriptRun.Interpreter == osconfigpb.SoftwareRecipe_Step_RunScript_POWERSHELL {
 	var qargs []string
+	//if step.ScriptRun.Interpreter == osconfigpb.SoftwareRecipe_Step_RunScript_POWERSHELL {
 	for _, arg := range step.ScriptRun.Args {
 		qargs = append(qargs, fmt.Sprintf("%q", arg))
 	}
-	res = append(res, "/tmp/scriptrun"+" "+strings.Join(qargs, " "))
-	return res, nil
+	command := "/tmp/scriptrun" + " " + strings.Join(qargs, " ")
+	args := []string{"-c", command}
+	return executeCommand("/bin/sh", filepath.Join(runDir, "stepName"), runEnvs, args...)
```

Comment:
if we name them all as 'stepName' from my original draft, multiple steps may end up using the same directory

i think the dir should be named as the comment above, i.e. `${working_dir}/${recipe_name}/step1_ScriptRun`

the stepN does mean we need to impart that information somehow, but is pretty important to avoid collision if the same step type repeats in a recipe.

Reply:
Yeah, I just used the same placeholders that were there. I removed the placeholders for runId and stepId now.